docs/pages/reference/introduction.mdx

```diff
@@ -23,7 +23,7 @@ running Teleport.
 - [Backends](./backends.mdx): Configuration options and required setup steps for
   Teleport's Auth Service storage backends.
 - [Join Methods](./join-methods.mdx): A reference of Teleport join methods and the token resources that configure them.
-- [Terraform Provider](./terraform-provider.mdx): Configuration options and
+- [Terraform Provider](./terraform/index.mdx): Configuration options and
```

Comment:
We'll need to use another file name than `index.mdx`, since the docs engine we're moving to doesn't support `index.mdx` files besides the one at the root of the docs site.

Reply:
Gotcha, I can easily change the data-source and resource index.mdx, the main one will require doing code changes in the reference generator (or just an mv in the bash script rendering the docs 😅 ). I'll fix this after the documentation engine change.

Reply:
Done